includes/wikia/ExternalUser_Wikia.php

```diff
@@ -232,49 +232,46 @@ public function mapToUser() {
 	}
 
 	/**
+	 * Adds the User object to the shared database
+	 *
 	 * @param User $User
 	 * @param String $password
 	 * @param String $email
 	 * @param String $realname
 	 *
-	 * @return bool
+	 * @return bool success
 	 */
-	protected function addToDatabase( $User, $password, $email, $realname ) {
-		global $wgExternalSharedDB;
+	protected function addToDatabase( User &$User, $password, $email, $realname ) {
 		wfProfileIn( __METHOD__ );
 
-		if( wfReadOnly() ) { // Change to wgReadOnlyDbMode if we implement that
-			wfDebug( __METHOD__ . ": Tried to add user to the $wgExternalSharedDB database while in wgReadOnly mode! " . $User->getName() . " [ " . $User->getId() . " ] (that's bad... fix the calling code)\n" );
-		} else {
-			wfDebug( __METHOD__ . ": add user to the $wgExternalSharedDB database: " . $User->getName() . " [ " . $User->getId() . " ] \n" );
+		$User->setToken();
 
-			$dbw = wfGetDB( DB_MASTER, array(), $wgExternalSharedDB );
-			$User->setToken();
+		global $wgExternalSharedDB;
+		$dbw = wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
 
+		try {
 			$dbw->insert(
 				'`user`',
-				array(
+				[
 					'user_id' => null,
 					'user_name' => $User->mName,
 					'user_password' => $User->mPassword,
 					'user_newpassword' => $User->mNewpassword,
-					'user_newpass_time' => $dbw->timestamp( $User->mNewpassTime ),
+					'user_newpass_time' => $dbw->timestamp($User->mNewpassTime),
```

Comment:
Actually, our current style guidelines include spaces around function args; here and L262,266